### PR TITLE
Trivialize rules

### PR DESCRIPTION
**Pull request by Hainish**
This PR sources the script by @RReverser from #4311, adding it to utils/ and running it on the ruleset.

Since #4311 contained so many ruleset changes, we decided it would be best to audit the script to ensure proper behavior, and re-run it against the latest ruleset.  I have done so, and noticed no unusual behavior after manually checking dozens of ruleset changes.

### CODE REVIEW

**Comment by Hainish** [comment]
There are numerous rules that have duplicate targets after running the script, I'll have to clean this up before merging.

**Comment by Hainish** [comment]
I've solved most of the validation problems by looking for duplicates only within the scope of a specific platform.  There were two remaining hosts after this change, which I've added to the duplicate-whitelist.txt

**Comment by Hainish** [comment]
@jsha can you review this PR (since it includes more than just the changes by @RReverser) and let me know if it looks good?  If it does, I can run the script over the ruleset once more and it won't cause failures in the test script.

**Comment by RReverser** [comment]
> Since #4311 contained so many ruleset changes, we decided it would be best to audit the script to ensure proper behavior, and re-run it against the latest ruleset. I have done so, and noticed no unusual behavior after manually checking dozens of ruleset changes.

Thanks! Although, as I said, I could do that myself as well. Would be nice to have that contribution bump on Github as an additional perk :)

> There are numerous rules that have duplicate targets after running the script, I'll have to clean this up before merging.

This is interesting, I thought this was handled by script.

File: utils/make-sqlite.py [hunk 1]
```diff
@@ -37,7 +37,12 @@
 c.execute('''DROP TABLE IF EXISTS targets''')
 c.execute('''CREATE TABLE targets
              (host TEXT,
+              platform_id INTEGER,
               ruleset_id INTEGER)''')
+c.execute('''DROP TABLE IF EXISTS platforms''')
+c.execute('''CREATE TABLE platforms
```
Comment (jsha) [inline utils/make-sqlite.py]: It's weird to add new code that depends on SQLite when we are planning to remove SQLite altogether. Can you do this in-memory in trivial-validate.py instead?

**Comment by jsha** [comment]
> Would be nice to have that contribution bump on Github profile as an additional perk

That should be fine - the main thing is that, given a pull request too large to review, we need some way to verify it. One way we could do it: Once this change is landed, you run the script and put up a PR. Bill runs the script is his repo too, pulls down your branch, and verifies no diffs. Assuming that's all good, Bill can consider the PR reviewed and merge it, which will give you credit for the commit.

**Comment by Hainish** [comment]
@RReverser perhaps an easier way to do it is to use `--author` in git, and attribute the ruleset diff to you in a separate commit?
cc @jsha

File: utils/make-sqlite.py (hunk 1 above)
Comment (Hainish) [inline utils/make-sqlite.py]: I was afraid you might say that :)
I figured I would just refactor this when the time came to get rid of `sqlite`, but I agree it adds complexity to something we're depricating. Refactoring it now.

**Comment by Hainish** [comment]
@jsha ready for review again.  I've removed all sqlite references from `trivial-validator.sh` but I wasn't sure if the `sqlite` db was used elsewhere, so I kept it in `utils/make-sqlite.sh`

**Comment by RReverser** [comment]
@Hainish Dunno if easier, never used that command so not sure how it exactly works. I assume you didn't change JS itself? If so, given that script is already reviewed, wouldn't be easier (and with the same result) if I would just update my PR after you merge your own changes separately?

**Comment by Hainish** [comment]
This is PR is now dependent on https://github.com/EFForg/https-everywhere/pull/4387, which I've separated out for clarity.

**Comment by Hainish** [comment]
@jsha rebased against master!

**Comment by jsha** [comment]
Mostly LGTM. The one comment I had: We should check for target hosts that don't get rewritten by any rule, and therefore get deleted. Mostly those will be errors in the rules, but there's a small possibility that those hosts may be affected by a securecookie tag. Talked to @Hainish in person and he's checking.

**Comment by RReverser** [comment]
Good point. Mostly I checked for the opposite as that was easy to do in scope of this PR, but that one would be easy as well.

**Comment by Hainish** [comment]
@jsha I didn't find any target domains of this type.  I double-checked by manually adding `foo.com` as a target domain to one of the overwritten files, and that's the only domain that is returned.  I wonder if someone manually went through and removed targets not affected by rewrite rules at some point?

**Comment by Hainish** [comment]
@RReverser feel free to update your PR with the results from the script, and I can compare to my own run & merge.

**Comment by RReverser** [comment]
@Hainish Awesome, will do. Thanks.

**Pull request merged**